### DISCUSSION TITLE
webviews of extensions (markdown preview, thunder client, etc) not being served from localhost?

### DISCUSSION THREAD

**Discussion by riexn** [gitpod-io/openvscode-server#335] [Q&A] [answered]
so this problem is happening because I am using openvscode in an offline environment.

I have two instances of openvscode-server, versions:
* 1.62.3
* 1.66.0

In version 1.62.3, I was able get the webviews of the extensions (markdown, thunder client, jupyter), where the pages were served from localhost:

![image](https://user-images.githubusercontent.com/37214468/162130048-5613edab-3a4d-4504-8d21-f37f08a42ae0.png)

in version 1.66.0, the webview request seems to be coming from an online service? so when I go offline, I can't view any of the webviews like so:

![image](https://user-images.githubusercontent.com/37214468/162131945-7c631a06-9a3a-4e39-9c38-c9a31e957718.png)

inspecting the network, it seems to make a request to an online service, but I am not sure why it was changed to that one rather than using localhost.

is it a misconfiguration from my side?

**Comment by jeanp413** [comment, marked as answer]
Hi @riexn, that change was intended to reduce the number of custom changes done by ourselves and align to upstream behavior, in the last release 1.66 they added [additional checks](https://github.com/microsoft/vscode/blob/3a90a6a46d95a523a210d28e42e24e14c28248e2/src/vs/workbench/contrib/webview/browser/pre/main.js#L342-L347) so it's no longer possible to serve from localhost directly

**Reply by riexn** [reply]
ahh, that's annoying.

**Reply by woshilaiba** [reply]
> Hi @riexn, that change was intended to reduce the number of custom changes done by ourselves and align to upstream behavior, in the last release 1.66 they added [additional checks](https://github.com/microsoft/vscode/blob/3a90a6a46d95a523a210d28e42e24e14c28248e2/src/vs/workbench/contrib/webview/browser/pre/main.js#L342-L347) so it's no longer possible to serve from localhost directly

May I ask why the vscode can load them when in offline mode ( no internet connection)?
